src/map/unit.c

```diff
@@ -3001,11 +3001,16 @@ static int unit_free(struct block_list *bl, enum clr_type clrtype)
 			VECTOR_CLEAR(sd->channels);
 			VECTOR_CLEAR(sd->script_queues);
 			VECTOR_CLEAR(sd->achievement); // Achievement [Smokexyz/Hercules]
-			VECTOR_CLEAR(sd->storage.item);
 			VECTOR_CLEAR(sd->hatEffectId);
 			VECTOR_CLEAR(sd->title_ids); // Title [Dastgir/Hercules]
 			VECTOR_CLEAR(sd->agency_requests);
-			sd->storage.received = false;
+
+			for (int i = 0; i < VECTOR_LENGTH(sd->storage.list); i++) { // Storages
+				VECTOR_CLEAR(VECTOR_INDEX(sd->storage.list, i).item);
+				VECTOR_INDEX(sd->storage.list, i).received = false;
```

Comment:
(adding here since the line is gone from the final diff)
the vector still needs to be cleared up, because its internal data is dynamically allocated. you just don't have to set `received` to false, as this data will be cleared up at `VECTOR_CLEAR(sd->storage.list);`